src/xmainloop.cpp

```diff
@@ -346,6 +364,8 @@ void XMainLoop::enternotify(XCrossingEvent* ce) {
         // https://tronche.com/gui/x/xlib/events/window-entry-exit/normal.html
         return;
     }
+    // Warning: we have to set this to false again!
+    duringEnterNotify_ = true;
```

Comment:
Here, I wasn't sure whats cleaner:
* a warning stating that we read if we add additional `return` instructions
* creating a kind of `OnScopeLeave` callback object that sets `duringEnterNotify_=false` in its destructor. This is more complicated, but works if we decide to add `return` statements somewhere in the following `if`-`else`-blocks.